plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/QueryBuilder.java

```diff
@@ -63,17 +64,27 @@ public QueryBuilder(JdbcClient client)
     public PreparedQuery prepareQuery(
             ConnectorSession session,
             Connection connection,
-            RemoteTableName remoteTableName,
+            Relation baseRelation,
             Optional<List<List<JdbcColumnHandle>>> groupingSets,
             List<JdbcColumnHandle> columns,
             TupleDomain<ColumnHandle> tupleDomain,
             Optional<String> additionalPredicate)
     {
-        String sql = "SELECT " + getProjection(columns);
-        sql += " FROM " + getRelation(remoteTableName);
-
         ImmutableList.Builder<QueryParameter> accumulator = ImmutableList.builder();
 
+        String sql = "SELECT " + getProjection(columns);
+        if (baseRelation instanceof Table) {
+            sql += " FROM " + getRelation(((Table) baseRelation).getName());
+        }
+        else if (baseRelation instanceof Query) {
+            PreparedQuery preparedQuery = ((Query) baseRelation).getPreparedQuery();
+            sql += " FROM (" + preparedQuery.getQuery() + ") o";
```

Comment:
can `o` here raise name conflicts?

Reply:
i don't think it can, that's why i used a fixed thing.